src/client/core/aad/auth-provider.ts

```diff
@@ -49,12 +62,18 @@ export default class AuthProvider {
         const client = this._getClient(tenantId);
 
         const authRequest = this._authRequest(resourceURI, tenantId);
-        if (this._accounts[tenantId]) {
+        try {
+            log.debug(`Trying to silently acquire token for '${tenantId}'`);
+            const account = await this._getAccount(tenantId);
+            if (!account) {
+                throw "no account";
```

Comment:
Let's make this error message more descriptive. Off the top of my head:

`Unable to find a valid AAD account for tenant ${tenantId}`

Also, since we're just throwing an error here anyway, let's switch this._getAccount() to throwing instead of returning null. No need to return null if we're not going to handle it gracefully.

Reply:
Good point about having `_getAccount()` throw. I just wanted auth to proceed the same way (with auth code flow) whether no account is found or the account is invalid (thrown by MSAL when no valid tokens are associated with the account).